app/routes.py

```diff
@@ -1,2 +1,235 @@
-from flask import Blueprint
+from flask import current_app, Blueprint, make_response, jsonify, request, Response
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+import requests
+from datetime import datetime
+import os
+from dotenv import load_dotenv
+import re
 
+load_dotenv()
+
+PATH = "https://slack.com/api/chat.postMessage"
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def post_task():
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return make_response({
+            "details": "Invalid data"
+        }, 400)
+
+    else: 
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        return make_response({
+            "task":{
+                "id": new_task.id,
+                "title": new_task.title,
+                "description": new_task.description,
+                "is_complete": bool(new_task.completed_at)
+            }
+        }, 201)
+
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def get_all_tasks():
+    if request.args.get('sort'):
+        if request.args.get('sort') == "asc":
+            tasks = Task.query.order_by(Task.title.asc()).all()
+        elif request.args.get('sort') == "desc":
+            tasks= Task.query.order_by(Task.title.desc()).all()
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append({
+            "id": task.id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": bool(task.completed_at)
+        })
+
+    return make_response(jsonify(tasks_response), 200)
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"], strict_slashes=False)
+def handle_one_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
```

Comment:
we could shorten this up:

```suggestion
    task = Task.query.get_or_404(task_id)
```